src/ObjectUtils/DwarfUtilsTest.cpp

```diff
@@ -0,0 +1,268 @@
+
+TEST(DwarfTypeAsString, DiesIfInvalid) {
+  FakeDWARFDie invalid_die;
+  invalid_die.is_valid_ = false;
+
+  EXPECT_DEATH((void)DwarfTypeAsString(invalid_die), "");
+}
+
+TEST(DwarfTypeAsString, ReturnsNameOnNamedTypes) {
+  EXPECT_STREQ(DwarfTypeAsString(kBaseTypeDie).data(), "int");
+  EXPECT_STREQ(DwarfTypeAsString(kUserSpecifiedType).data(), "Foo");
+}
+
+TEST(DwarfTypeAsString, PrependsAtomicModifier) {
+  FakeDWARFDie atomic_die;
+  atomic_die.tag_ = llvm::dwarf::DW_TAG_atomic_type;
+  atomic_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(atomic_die).data(), "int _Atomic");
+}
+
+TEST(DwarfTypeAsString, PrependsConstModifier) {
+  FakeDWARFDie const_die;
+  const_die.tag_ = llvm::dwarf::DW_TAG_const_type;
+  const_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(const_die).data(), "int const");
+}
+
+TEST(DwarfTypeAsString, PrependsVolatileModifier) {
+  FakeDWARFDie volatile_die;
+  volatile_die.tag_ = llvm::dwarf::DW_TAG_volatile_type;
+  volatile_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(volatile_die).data(), "int volatile");
+}
+
+TEST(DwarfTypeAsString, PrependsRestrictModifier) {
+  FakeDWARFDie restrict_die;
+  restrict_die.tag_ = llvm::dwarf::DW_TAG_restrict_type;
+  restrict_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(restrict_die).data(), "int restrict");
+}
+
+TEST(DwarfTypeAsString, AppendsArrayType) {
+  FakeDWARFDie array_die;
+  array_die.tag_ = llvm::dwarf::DW_TAG_array_type;
+  array_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(array_die).data(), "int[]");
+}
+
+TEST(DwarfTypeAsString, AppendsPointerType) {
+  FakeDWARFDie pointer_die;
+  pointer_die.tag_ = llvm::dwarf::DW_TAG_pointer_type;
+  pointer_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(pointer_die).data(), "int*");
+}
+
+TEST(DwarfTypeAsString, AppendsReferenceType) {
+  FakeDWARFDie reference_die;
+  reference_die.tag_ = llvm::dwarf::DW_TAG_reference_type;
+  reference_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(reference_die).data(), "int&");
+}
+
+TEST(DwarfTypeAsString, AppendsRValueReferenceType) {
+  FakeDWARFDie rvalue_reference_die;
+  rvalue_reference_die.tag_ = llvm::dwarf::DW_TAG_rvalue_reference_type;
+  rvalue_reference_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(rvalue_reference_die).data(), "int&&");
+}
+
+TEST(DwarfTypeAsString, PointerToConstIntVsConstPointerToInt) {
+  FakeDWARFDie pointer_die;
+  pointer_die.tag_ = llvm::dwarf::DW_TAG_pointer_type;
+  pointer_die.type_ = &kBaseTypeDie;
+
+  FakeDWARFDie const_die;
+  const_die.tag_ = llvm::dwarf::DW_TAG_const_type;
+  const_die.type_ = &pointer_die;
+
+  EXPECT_STREQ(DwarfTypeAsString(const_die).data(), "int* const");
+
+  // Let's switch parent/child and expect a const pointer to an int.
+  pointer_die.type_ = &const_die;
+  const_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(pointer_die).data(), "int const*");
+}
```

Comment:
The two examples are reversed in respect to the test name and the comment. `int* const` is the const pointer to int and `int const*` is the pointer to const int.

Reply:
Thanks. Should be correct now.